fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAlterMetaJob.java

```diff
@@ -0,0 +1,497 @@
+                    }
+                }
+                errMsg += ". This operation maybe partial successfully, You should retry until success.";
+                LOG.warn(errMsg);
+                throw new DdlException(errMsg);
+            }
+        }
+    }
+
+    void updateNextVersion(@NotNull LakeTable table) {
+        for (long partitionId : partitionIndexMap.rowKeySet()) {
+            Partition partition = table.getPartition(partitionId);
+            long commitVersion = commitVersionMap.get(partitionId);
+            Preconditions.checkState(partition.getNextVersion() == commitVersion,
+                    "partitionNextVersion=" + partition.getNextVersion() + " commitVersion=" + commitVersion);
+            partition.setNextVersion(commitVersion + 1);
+            LOG.info("partitionNextVersion=" + partition.getNextVersion() + " commitVersion=" + commitVersion);
+        }
+    }
+
+    void updateVisibleVersion(@NotNull LakeTable table) {
+        for (long partitionId : partitionIndexMap.rowKeySet()) {
+            Partition partition = table.getPartition(partitionId);
+            long commitVersion = commitVersionMap.get(partitionId);
+            Preconditions.checkState(partition.getVisibleVersion() == commitVersion - 1,
+                    "partitionVisitionVersion=" + partition.getVisibleVersion() + " commitVersion=" + commitVersion);
+            partition.updateVisibleVersion(commitVersion);
+            LOG.info("partitionVisibleVersion=" + partition.getVisibleVersion() + " commitVersion=" + commitVersion);
+        }
+    }
+
+    @Override
+    protected boolean cancelImpl(String errMsg) {
+        if (jobState == JobState.CANCELLED || jobState == JobState.FINISHED) {
+            return false;
+        }
+
+        Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+        if (db == null) {
+            // database has been dropped
+            return false;
+        }
+        db.writeLock();
+        LakeTable table;
+        try {
+            table = (LakeTable) db.getTable(tableId);
+
+            // Cancel a job of state `FINISHED_REWRITING` only when the database or table has been dropped.
+            if (jobState == JobState.FINISHED_REWRITING && table != null) {
+                return false;
+            }
+
+            this.jobState = JobState.CANCELLED;
+            this.errMsg = errMsg;
+            this.finishedTimeMs = System.currentTimeMillis();
+
+            if (table != null) {
+                table.setState(OlapTable.OlapTableState.NORMAL);
+            }
+
+            if (span != null) {
+                span.setStatus(StatusCode.ERROR, errMsg);
+                span.end();
+            }
+        } finally {
+            db.writeUnlock();
+        }
+
+        GlobalStateMgr.getCurrentState().getEditLog().logAlterJob(this);
+        return true;
+    }
+
+    @Override
+    protected void getInfo(List<List<Comparable>> infos) {
+
```

Comment:
Why implement an empty function? Please complete the implementation, or add a comment explaining why there is an empty function body here.

Reply:
LakeTableAlterMetaJob is not supported by show for now